ast/annotations.go

```diff
@@ -69,15 +70,19 @@ type (
 	}
 
 	AnnotationsRef struct {
-		Location    *Location    `json:"location"` // The location of the node the annotations are applied to
-		Path        Ref          `json:"path"`     // The path of the node the annotations are applied to
+		Path        Ref          `json:"path"` // The path of the node the annotations are applied to
 		Annotations *Annotations `json:"annotations,omitempty"`
-		node        Node         // The node the annotations are applied to
+		Location    *Location    `json:"location,omitempty"` // The location of the node the annotations are applied to
+
+		node Node // The node the annotations are applied to
 	}
 )
 
 func (a *Annotations) String() string {
-	bs, _ := json.Marshal(a)
+	opts := NewDefaultMarshalJSONOptions()
+	opts.Locations = false
+
+	bs, _ := MarshalJSON(a, opts)
```

Comment:
This isn't really good enough since it will remove the location information during marshalling. I'm still trying to work out what to do here...

Reply:
Since we are now changing the `-` to `location,omitempty` in the node structs, we'll have to change each of their `String` methods too or is only `Annotations` affected? This also means nodes will have location info which is new behavior.

Also I wonder if we can pass these new options via `ast.ParserOptions`.

Reply:
👍 I think this is doable. I've had to write some custom marshalling functions for the last types, but this makes the most sense.